connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceDefinition.java

```diff
@@ -166,6 +168,8 @@ static void registerTransformers120(ResourceTransformationDescriptionBuilder par
         builder.getAttributeBuilder()
                 .setDiscard(new DiscardAttributeChecker.DiscardAttributeValueChecker(new ModelNode(false)), CONNECTABLE)
                 .setDiscard(new DiscardAttributeChecker.DiscardAttributeValueChecker(false, false, new ModelNode(true)), STATISTICS_ENABLED)
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, ENLISTMENT_TRACE, MCP)
```

Comment:
AIUI, the hard-coded behavior for ENLISTMENT_TRACE in previous releases was equivalent to 'true'. So if the value is 'true' it can be discarded.

If there's a similar thing for MCP (a class name that was the hard coded class used before) then the same thing can be done for MCP.